torch/_dynamo/external_utils.py

```diff
@@ -97,6 +97,21 @@ def untyped_storage_size(x: torch.Tensor):
     return x.untyped_storage().size()
 
 
+class CompiledAutogradEngine:
+    @staticmethod
+    def queue_callback(final_callbacks, cb):
+        final_callbacks.append(cb)
+
+    @staticmethod
+    def exec_final_callbacks(final_callbacks):
+        for cb in final_callbacks:
+            cb()
```

Comment:
As discussed in the test above, adding more callbacks from within a callback is expected to be supported. Are we explicitly saying we don't want to support this here?

Reply:
it seems that Dynamo doesn't support for-loop dynamically adding element to the list being looped over yet. I'll add a todo comment for this

Reply:
I iterated more on it and found a way to make it work :)